staticcheck/lint.go

```diff
@@ -3912,3 +3912,30 @@ func CheckStaticBitShift(pass *analysis.Pass) (interface{}, error) {
 
 	return nil, nil
 }
+
+var builtinLessThanZeroQ = pattern.MustParse(`
+	(Or
+		(BinaryExpr
+			(BasicLit "INT" "0")
+			"<"
```

Comment:
I believe this should be `>`, not `<` – len and cap can very well be larger than 0. 0 can't be larger than len and cap. The same mistake exists in your unit tests.

Reply:
D'oh! Copy–paste strikes again!
Fixed.